packages/react/src/components/DatePicker/DatePicker.tsx

```diff
@@ -653,7 +661,7 @@ const DatePicker = React.forwardRef(function DatePicker(
           savedOnChange(...args);
         }
       },
-      onClose: savedOnClose,
+
```

Comment:
This is the only spot `savedOnClose` is used directly, but it's also used in the dependency array of the huge useEffect this is all wrapped in.